pegasus/sites.v3/code.org/public/educate/weblab-test.html

```diff
@@ -0,0 +1,217 @@
+                  <p class="incomplete">Not complete</p>
+                  <p class="loading" style="display: none">Connecting...</p>
+                  <p class="succeeded" style="display: none">Success</p>
+                  <p class="failed" style="display: none">Failed</p>
+                </td>
+              </tr>
+              <tr>
+                <td>Connected to <code>https://codeprojects.org</code></td>
+                <td class="codeprojects">
+                  <p class="incomplete">Not complete</p>
+                  <p class="loading" style="display: none">Connecting...</p>
+                  <p class="succeeded" style="display: none">Success</p>
+                  <p class="failed" style="display: none">Failed</p>
+                </td>
+              </tr>
+              <tr>
+                <td>Able to load an iFrame</td>
+                <td class="iframe">
+                  <p class="incomplete">Not complete</p>
+                  <p class="loading" style="display: none">Connecting...</p>
+                  <p class="succeeded" style="display: none">Success</p>
+                  <p class="failed" style="display: none">Failed</p>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+        <div>
+          <button id="connect" class="btn btn-primary">Connect</button>
+        </div>
+        <p id="success-message" style="display: none">
+          <strong>You passed all checks! No further action is needed.</strong>
+        </p>
+        <div id="fail-message" style="display: none">
+          <p>
+            <strong>Error. One or more checks failed.</strong>
+          </p>
+          <p>
+            <strong>Action required: </strong>
+            Please contact your IT department to update your school’s firewall
+            settings or device policies. Contact our Support Team (<a
+              href="mailto:support@code.org"
+              >support@code.org</a
+            >) if you have any questions.
+          </p>
+        </div>
+        <iframe id="test-iframe" src="/" style="display: none"></iframe>
+      </div>
+    </div>
+  </div>
+</div>
+<script>
+  const TIMEOUT_MS = 3000;
+  function testImageAccess(url, callback = () => {}) {
+    return new Promise((resolve, reject) => {
+      const element = new Image();
+      element.src = url;
+
+      let succeeded = false;
+      element.onerror = () => {
+        callback(new Error());
+        reject();
+      };
+      element.onload = () => {
+        succeeded = true;
+        callback();
+        resolve();
+      };
+      setTimeout(() => {
+        if (!succeeded) {
+          callback(new Error());
+          reject();
+        }
+      }, TIMEOUT_MS);
+
+      // store a reference to the element so it doesn't get collected
+      window.testImages = window.testImages || [];
+      window.testImages.push(element);
+    });
+  }
+
+  function testIFrameAccess(successCallback, failureCallback) {
```

Comment:
Are these arguments necessary?

Reply:
No, thank you! Leftover from when I was passing these functions as arguments (which is arguably better here, because it's more consistent with how the other two checks are working, but since I've been testing it like this I'll just remove these arguments for now).